src/Infrastructure/Rsp.IrasService.Infrastructure/Interceptors/AuditTrailInterceptor.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Rsp.IrasService.Domain.Entities;
+using Rsp.IrasService.Domain.Interfaces;
+using Rsp.IrasService.Infrastructure.Helpers;
+
+namespace Rsp.IrasService.Infrastructure.Interceptors;
+
+[ExcludeFromCodeCoverage]
+public class AuditTrailInterceptor(IAuditTrailDetailsService auditTrailDetailsService, IEnumerable<IAuditTrailHandler> auditTrailHandlers) : SaveChangesInterceptor
+{
+    public override async ValueTask<InterceptionResult<int>> SavingChangesAsync(
+        DbContextEventData eventData,
+        InterceptionResult<int> result,
+        CancellationToken cancellationToken = default)
```

Comment:
When using the parameters on new line use the following style, it makes it more readable.
```suggestion
    public override async ValueTask<InterceptionResult<int>> SavingChangesAsync
    (
        DbContextEventData eventData,
        InterceptionResult<int> result,
        CancellationToken cancellationToken = default
    )
```